JSON parse error with latest docker image (2024-12-20)

When I update with latest docker image (2024-12-20), I got this error in logs :

```
2024-12-26T01:01:49.666Z  INFO 1 --- [           main] c.d.t.e.TadoExporterApplicationKt        : Started TadoExporterApplicationKt in 8.542 seconds (process running for 9.607)
2024-12-26T01:01:49.678Z  INFO 1 --- [   scheduling-1] c.d.t.e.metrics.HomeModelRefresher       : Refreshing zones for all known homes.
2024-12-26T01:01:49.678Z  INFO 1 --- [   scheduling-1] c.d.t.e.metrics.HomeModelRefresher       : Initializing homes from API.
2024-12-26T01:01:49.925Z  INFO 1 --- [   scheduling-1] c.d.t.e.apiclient.auth.TadoAuthFilter    : Obtaining new bearer token for adrien.yhuel@gmail.com.
2024-12-26T01:01:50.815Z  INFO 1 --- [   scheduling-1] c.d.t.exporter.metrics.TadoMeterFactory  : Adding gauges for weather information for home 'Maison' (518383)
2024-12-26T01:01:50.829Z  INFO 1 --- [   scheduling-1] c.d.t.e.metrics.HomeModelRefresher       : 1 homes initialized.
2024-12-26T01:01:50.830Z  INFO 1 --- [   scheduling-1] c.d.t.e.metrics.HomeModelRefresher       : Refreshing zones for home 'Maison' (518383).
2024-12-26T01:01:51.387Z ERROR 1 --- [   scheduling-1] o.s.s.s.TaskUtils$LoggingErrorHandler    : Unexpected error occurred in scheduled task
org.springframework.web.client.RestClientException: Error while extracting response for type [class click.dobel.tado.exporter.apiclient.model.Zones] and content type [application/json;charset=UTF-8]
	at org.springframework.web.client.HttpMessageConverterExtractor.extractData(HttpMessageConverterExtractor.java:127) ~[spring-web-6.2.1.jar!/:6.2.1]
	at org.springframework.web.client.RestTemplate.doExecute(RestTemplate.java:903) ~[spring-web-6.2.1.jar!/:6.2.1]
	at org.springframework.web.client.RestTemplate.execute(RestTemplate.java:801) ~[spring-web-6.2.1.jar!/:6.2.1]
	at org.springframework.web.client.RestTemplate.getForObject(RestTemplate.java:415) ~[spring-web-6.2.1.jar!/:6.2.1]
	at click.dobel.tado.exporter.apiclient.TadoApiClient.zones(TadoApiClient.kt:88) ~[!/:6.3.0]
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:na]
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[na:na]
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[na:na]
	at java.base/java.lang.reflect.Method.invoke(Unknown Source) ~[na:na]
	at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:359) ~[spring-aop-6.2.1.jar!/:6.2.1]
	at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:196) ~[spring-aop-6.2.1.jar!/:6.2.1]
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163) ~[spring-aop-6.2.1.jar!/:6.2.1]
	at org.springframework.cache.interceptor.CacheInterceptor.lambda$invoke$0(CacheInterceptor.java:55) ~[spring-context-6.2.1.jar!/:6.2.1]
	at org.springframework.cache.interceptor.CacheAspectSupport.invokeOperation(CacheAspectSupport.java:430) ~[spring-context-6.2.1.jar!/:6.2.1]
	at org.springframework.cache.interceptor.CacheAspectSupport.lambda$executeSynchronized$2(CacheAspectSupport.java:468) ~[spring-context-6.2.1.jar!/:6.2.1]
	at org.springframework.cache.caffeine.CaffeineCache$LoadFunction.apply(CaffeineCache.java:237) ~[spring-context-support-6.2.1.jar!/:6.2.1]
	at com.github.benmanes.caffeine.cache.BoundedLocalCache.lambda$doComputeIfAbsent$14(BoundedLocalCache.java:2688) ~[caffeine-3.1.8.jar!/:3.1.8]
	at java.base/java.util.concurrent.ConcurrentHashMap.compute(Unknown Source) ~[na:na]
	at com.github.benmanes.caffeine.cache.BoundedLocalCache.doComputeIfAbsent(BoundedLocalCache.java:2686) ~[caffeine-3.1.8.jar!/:3.1.8]
	at com.github.benmanes.caffeine.cache.BoundedLocalCache.computeIfAbsent(BoundedLocalCache.java:2669) ~[caffeine-3.1.8.jar!/:3.1.8]
	at com.github.benmanes.caffeine.cache.LocalCache.computeIfAbsent(LocalCache.java:112) ~[caffeine-3.1.8.jar!/:3.1.8]
	at com.github.benmanes.caffeine.cache.LocalManualCache.get(LocalManualCache.java:62) ~[caffeine-3.1.8.jar!/:3.1.8]
	at org.springframework.cache.caffeine.CaffeineCache.get(CaffeineCache.java:135) ~[spring-context-support-6.2.1.jar!/:6.2.1]
	at org.springframework.cache.interceptor.AbstractCacheInvoker.doGet(AbstractCacheInvoker.java:97) ~[spring-context-6.2.1.jar!/:6.2.1]
	at org.springframework.cache.interceptor.CacheAspectSupport.executeSynchronized(CacheAspectSupport.java:468) ~[spring-context-6.2.1.jar!/:6.2.1]
	at org.springframework.cache.interceptor.CacheAspectSupport.execute(CacheAspectSupport.java:437) ~[spring-context-6.2.1.jar!/:6.2.1]
	at org.springframework.cache.interceptor.CacheAspectSupport.execute(CacheAspectSupport.java:409) ~[spring-context-6.2.1.jar!/:6.2.1]
	at org.springframework.cache.interceptor.CacheInterceptor.invoke(CacheInterceptor.java:65) ~[spring-context-6.2.1.jar!/:6.2.1]
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:184) ~[spring-aop-6.2.1.jar!/:6.2.1]
	at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:97) ~[spring-aop-6.2.1.jar!/:6.2.1]
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:184) ~[spring-aop-6.2.1.jar!/:6.2.1]
	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:727) ~[spring-aop-6.2.1.jar!/:6.2.1]
	at click.dobel.tado.exporter.apiclient.TadoApiClient$$SpringCGLIB$$0.zones(<generated>) ~[!/:6.3.0]
	at click.dobel.tado.exporter.metrics.HomeModelRefresher.refreshHomeModel(HomeModelRefresher.kt:43) ~[!/:6.3.0]
	at click.dobel.tado.exporter.metrics.HomeModelRefresherSchedule.scheduleRefresh(HomeModelRefresherSchedule.kt:14) ~[!/:6.3.0]
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:na]
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[na:na]
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[na:na]
	at java.base/java.lang.reflect.Method.invoke(Unknown Source) ~[na:na]
	at org.springframework.scheduling.support.ScheduledMethodRunnable.runInternal(ScheduledMethodRunnable.java:130) ~[spring-context-6.2.1.jar!/:6.2.1]
	at org.springframework.scheduling.support.ScheduledMethodRunnable.lambda$run$2(ScheduledMethodRunnable.java:124) ~[spring-context-6.2.1.jar!/:6.2.1]
	at io.micrometer.observation.Observation.observe(Observation.java:498) ~[micrometer-observation-1.14.2.jar!/:1.14.2]
	at org.springframework.scheduling.support.ScheduledMethodRunnable.run(ScheduledMethodRunnable.java:124) ~[spring-context-6.2.1.jar!/:6.2.1]
	at org.springframework.scheduling.config.Task$OutcomeTrackingRunnable.run(Task.java:85) ~[spring-context-6.2.1.jar!/:6.2.1]
	at org.springframework.scheduling.support.DelegatingErrorHandlingRunnable.run(DelegatingErrorHandlingRunnable.java:54) ~[spring-context-6.2.1.jar!/:6.2.1]
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) ~[na:na]
	at java.base/java.util.concurrent.FutureTask.runAndReset(Unknown Source) ~[na:na]
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source) ~[na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) ~[na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) ~[na:na]
	at java.base/java.lang.Thread.run(Unknown Source) ~[na:na]
Caused by: org.springframework.http.converter.HttpMessageNotReadableException: JSON parse error: Cannot construct instance of `click.dobel.tado.exporter.apiclient.model.Zones` (although at least one Creator exists): no default no-arguments constructor found
	at org.springframework.http.converter.json.AbstractJackson2HttpMessageConverter.readJavaType(AbstractJackson2HttpMessageConverter.java:409) ~[spring-web-6.2.1.jar!/:6.2.1]
	at org.springframework.http.converter.json.AbstractJackson2HttpMessageConverter.read(AbstractJackson2HttpMessageConverter.java:357) ~[spring-web-6.2.1.jar!/:6.2.1]
	at org.springframework.web.client.HttpMessageConverterExtractor.extractData(HttpMessageConverterExtractor.java:105) ~[spring-web-6.2.1.jar!/:6.2.1]
	... 50 common frames omitted
Caused by: com.fasterxml.jackson.databind.exc.MismatchedInputException: Cannot construct instance of `click.dobel.tado.exporter.apiclient.model.Zones` (although at least one Creator exists): no default no-arguments constructor found
 at [Source: REDACTED (`StreamReadFeature.INCLUDE_SOURCE_IN_LOCATION` disabled); line: 1, column: 1]
	at com.fasterxml.jackson.databind.exc.MismatchedInputException.from(MismatchedInputException.java:63) ~[jackson-databind-2.18.2.jar!/:2.18.2]
	at com.fasterxml.jackson.databind.DeserializationContext.reportInputMismatch(DeserializationContext.java:1754) ~[jackson-databind-2.18.2.jar!/:2.18.2]
	at com.fasterxml.jackson.databind.DeserializationContext.handleMissingInstantiator(DeserializationContext.java:1379) ~[jackson-databind-2.18.2.jar!/:2.18.2]
	at com.fasterxml.jackson.databind.deser.ValueInstantiator.createUsingDefault(ValueInstantiator.java:248) ~[jackson-databind-2.18.2.jar!/:2.18.2]
	at com.fasterxml.jackson.databind.deser.std.StdValueInstantiator.createUsingDefault(StdValueInstantiator.java:275) ~[jackson-databind-2.18.2.jar!/:2.18.2]
	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.createDefaultInstance(CollectionDeserializer.java:264) ~[jackson-databind-2.18.2.jar!/:2.18.2]
	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:246) ~[jackson-databind-2.18.2.jar!/:2.18.2]
	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:30) ~[jackson-databind-2.18.2.jar!/:2.18.2]
	at com.fasterxml.jackson.databind.deser.DefaultDeserializationContext.readRootValue(DefaultDeserializationContext.java:342) ~[jackson-databind-2.18.2.jar!/:2.18.2]
	at com.fasterxml.jackson.databind.ObjectReader._bindAndClose(ObjectReader.java:2125) ~[jackson-databind-2.18.2.jar!/:2.18.2]
	at com.fasterxml.jackson.databind.ObjectReader.readValue(ObjectReader.java:1501) ~[jackson-databind-2.18.2.jar!/:2.18.2]
	at org.springframework.http.converter.json.AbstractJackson2HttpMessageConverter.readJavaType(AbstractJackson2HttpMessageConverter.java:398) ~[spring-web-6.2.1.jar!/:6.2.1]
	... 52 common frames omitted
```

--- a/tado-exporter/src/main/kotlin/click/dobel/tado/exporter/apiclient/model/Zones.kt
+++ b/tado-exporter/src/main/kotlin/click/dobel/tado/exporter/apiclient/model/Zones.kt
@@ -2,7 +2,8 @@ package click.dobel.tado.exporter.apiclient.model
 
 import click.dobel.tado.api.Zone
 import click.dobel.tado.exporter.metrics.ZoneEntry
+import com.fasterxml.jackson.annotation.JsonCreator
 
-class Zones(zones: Collection<Zone>) : List<Zone> by ArrayList(zones)
+class Zones @JsonCreator constructor(zones: Collection<Zone>) : List<Zone> by ArrayList(zones)
 
 fun Collection<Zone>.toEntrySet() = this.map(::ZoneEntry).toSet()

